strapi/strapi
# GraphQL Plugin (bug): Content type's attributes marked as private are being exposed in the exported GraphQL schema

### What does it do?

It fixes a bug in the strapi graphql plugin in the functions generating input  and types for graphql. The detection of an attribute as private was only applied to the privateAttributes array in the model.json file instead of also looking at the private property on an attribute within a model. There was also a bug in how the private attributes were filtered the attribute name was not being passed in the function so added the right filtering rule to only process fields that are not private and export them to the graphql schema file.

### Why is it needed?

This fix will remove all private attributes from the graphql schema as this data is not meant to be exposed to the outside world if marked as private by the user.

### How to test it?

In the content type builder, create an enum attribute and a string attribute as part of a content type and mark both as private and hit the save button. Open the graphql playground and click on the schema button on the right side of the screen and see if your enum and string value are not present in the exported schema.

### Related issue(s)/PR(s)

Fixes issue #9804
